### PR TITLE
A0-2975: add counters for errors

### PR DESCRIPTION
**Pull request by kostekIV**
# Description
Counters for errors, where possible. In handle_state_response and handle_request_response its hard to tell whether there was an error or success, so we count everything. 

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality

### CODE REVIEW

**Review by timorl — APPROVED** [review]
A single comment, but the thing about reporting totals+errors rather than successes+errors might take a while, if you think it should be done.

File: finality-aleph/src/sync/service.rs [hunk 1]
```diff
@@ -319,15 +338,14 @@ where
...
-        self.metrics.report_sync_handle_request();
 
         match self.handler.handle_request(request) {
             Ok(Action::Response(response_items)) => {
                 let mut limiter = MsgLimiter::new(&response_items);
                 loop {
                     match limiter.next_largest_msg() {
                         Ok(None) => {
-                            break self.metrics.report_sync_handle_request();
+                            break self.metrics.report_sync_handle_request_success();
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/sync/service.rs:348]: this is inside a loop, so we only report success if we end the loop 'normally'. I agree that counter for total calls would be nicer, I will do that

**Review by ggawryal — APPROVED** [review]
LGTM

File: finality-aleph/src/metrics.rs [hunk 2]
```diff
@@ -246,27 +294,51 @@ impl<H: Key> Metrics<H> {
         }
     }
 
-    pub fn report_sync_broadcast(&self) {
+    pub fn report_sync_broadcast_success(&self) {
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply finality-aleph/src/metrics.rs]: hmmmm, It would be nice but Im not sure if this is worth it :P

**Review by timorl — CHANGES_REQUESTED** [review]
I would prefer if @ggawryal 's comment got implemented, the code duplication now is pretty extensive and it was a very good idea for lowering it somewhat.

**Comment by kostekIV** [comment]
> I would prefer if @ggawryal 's comment got implemented, the code duplication now is pretty extensive and it was a very good idea for lowering it somewhat.

hmm,  Now code duplication is inside the file metrics.rs while with the enum solution, we will crowd the code in the service.rs with the additional `use  MetricsEnum::*` in every function. So for me, it's about having more duplications in the metrics code vs more boilerplate code in the service.rs and I prefer to have less of such code in a more critical place which is service.rs and the tradeoff makes it not worth it in my opinion to implement this.

I hate to say that but if we want to lower the code duplication in metrics.rs we could use some macro (ble)

**Comment by timorl** [comment]
I believe even if the comparison is `metrics.report_sync_event(SyncEvent::Broadcast)` vs `metrics.report_sync_broadcast()` the former still wins. You also get _much_ more readable code in `metrics.rs` -- you can have a pretty clear `match` over the enum variants mapping into the metric names, separate from the reporting code, easier consistency with the `_error` suffix, and probably more things. It's just so~ much better organized then. Dx

Oh, and if you don't agree with the first sentence, you can always introduce a `MetricEvent` trait and parametrize `report<E: MetricEvent>(&self, event: M)`, which will remove the double `sync_event` there and likely will be quite useful in the future. }:3

**Comment by kostekIV** [comment]
> I believe even if the comparison is `metrics.report_sync_event(SyncEvent::Broadcast)` vs `metrics.report_sync_broadcast()` the former still wins. You also get _much_ more readable code in `metrics.rs` -- you can have a pretty clear `match` over the enum variants mapping into the metric names, separate from the reporting code, easier consistency with the `_error` suffix, and probably more things. It's just so~ much better organized then. Dx
> 
> Oh, and if you don't agree with the first sentence, you can always introduce a `MetricEvent` trait and parametrize `report<E: MetricEvent>(&self, event: E)`, which will remove the double `sync_event` there and likely will be quite useful in the future. }:3

mhh, ok done :|

**Pull request merged**